### PR TITLE
Upgrades icon generation script to Python 3

### PR DESCRIPTION
**Pull request by dietmarw**

### CODE REVIEW

**Comment by dietmarw** [comment]
@hkiel Could you have a quick look?

**Comment by sjoelund** [comment]
Python 2/3 differences?

File: Examples/generate_icons.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def getGraphicsForClass(modelicaClass):
             graphicsObj['fillPattern'] = g[11]
             graphicsObj['lineThickness'] = float(g[12])
             graphicsObj['extent'] = [[float(g[13]), float(g[14])], [float(g[15]), float(g[16])]]
-            graphicsObj['textString'] = g[17].strip('"').encode('utf-8')
```
Comment (hkiel) [inline Examples/generate_icons.py]: Seems to work fine without encoding, the `decode()` there was wrong.
Why not remova also `encode()` in line 300?

**Comment by hkiel** [comment]
Without encode() output looks fine with 2.7. I checked `Modelica.ComplexBlocks.ComplexMath.Bode`

**Comment by hkiel** [comment]
The symlink in line 1137 could be protected by
```
if not os.path.exists(filename):
    os.symlink(hashName, filename)
else:
    logger.warning('Target file {0} already exists'.format(filename))
```
for dumb (= case insensitive) filesystems

File: Examples/generate_icons.py (hunk 1 above)
Comment (dietmarw) [reply Examples/generate_icons.py]: I overlooked that one. But there are two more and I'm not 100% sure of the impact. So perhaps somebody else can have a look at those.

**Comment by dietmarw** [comment]
> The symlink in line 1137 could be protected by
> 
> ```
> if not os.path.exists(filename):
>     os.symlink(hashName, filename)
> else:
>     logger.warning('Target file {0} already exists'.format(filename))
> ```
> for dumb (= case insensitive) filesystems

That would just warn but not solve the issue right?

**Comment by dietmarw** [comment]
I found another issue. A text string is not correctly scaled. See https://build.openmodelica.org/Documentation/Icons/Complex.svg
which really should look like:
image

**Comment by dietmarw** [comment]
You should be able to push to my PR.

**Comment by hkiel** [comment]
> > for dumb (= case insensitive) filesystems
> 
> That would just warn but not solve the issue right?

 Right. But you cannot do anything about it, and IMO it should not fail. At least you get a warning.

**Comment by hkiel** [comment]
> You should be able to push to my PR.

Cannot push: `remote: Permission to dietmarw/OMCompiler.git denied to hkiel`

**Comment by dietmarw** [comment]
OK in that case push this to your branch and I can cherry pick it from there.

**Comment by hkiel** [comment]
I created #2684 with the fixes for text height.

We should think about moving from deprected `optparse` to `argparse`

**Comment by dietmarw** [comment]
@sjoelund Could you merge this PR and #2684 after that?

**Comment by dietmarw** [comment]
@hkiel 	I'm always for updating deprecated code.

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2018-10-01_16-15-15](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6070/console).

**Pull request closed without merging**